chain/network/src/network_protocol/mod.rs

```diff
@@ -90,6 +90,45 @@ impl std::str::FromStr for PeerAddr {
     }
 }
 
+// Wrapper around std::net::IpAddr, which constructs a SignedOwnedIpAddress
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct OwnedIpAddress {
```

Comment:
As discussed offline, it's needed to implement the interface in order to hide the serialization of ip_addr from the user and in order to form the message (SignedOwnedIpAddress) to be sent over the network to contain both required ip address and signature

If i understand correctly, there always needs to be a wrapper class in order to introduce new member methods to existing class such as the examples below
- https://github.com/near/nearcore/blob/7d0b0b4207325f2dab2e77486341a87f5eb111d1/chain/network/src/tcp.rs#L150-L152
- https://github.com/near/nearcore/blob/7d0b0b4207325f2dab2e77486341a87f5eb111d1/chain/network/src/tcp.rs#L60
- https://github.com/near/nearcore/blob/7d0b0b4207325f2dab2e77486341a87f5eb111d1/chain/network/src/tcp.rs#L212

Reply:
Regarding 1. good point, I thought about it before but somehow forgot it. I guess I preferred the other approach at the time as I could refer to protobuf serialization of OwnedAccount. Updated to SignedIpAddress with constructors as suggested. 

Ignoring 2. since it's redundant thanks to 1.